File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.java
class Solution {
    static final int mod = 1000000007;
    public int xorAfterQueries(int[] nums, int[][] q) {
        int n = nums.length;
        int ans = 0;
        for(int[] vec : q){
            int l = vec[0];
            int r = vec[1];
            int k = vec[2];
            int v = vec[3];
            for(int i=l;i<=r;i+=k){
                nums[i]=(int)((nums[i]*1L*v)%mod);
            }
        }
        for(int i : nums){
            ans^=i;
        }
        return ans;
    }

}

File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.cpp
const int mod = 1e9+7;
class Solution {
public:
    int xorAfterQueries(vector<int>& nums, vector<vector<int>>& q) {
        int n = nums.size();
        int ans = 0;
        
        for(auto i:q){
            auto vec = i;
            int l = vec[0];
            int r = vec[1];
            int inc = vec[2];
            int val = vec[3];
            for(int j=l;j<=r;j+=inc){
                nums[j]=(nums[j]*1LL*val)%(mod);
            }
        }
        for(auto i:nums){
            ans^=i;
        }
        return ans;
    }

};